host_tools/fw_update.py

```diff
@@ -80,18 +102,26 @@ def scan_for_device(self):
         ports = [x[0] for x in list_ports.comports()]
         retport = None
         self._log("Searching for an active bootloader")
-        
+
         for p in ports:
             try:
                 rsp = self._send_ping(p)
                 if not rsp == None:
-                    self._log("Found bootloader on: {}".format(p))
+                    ver = self._get_version(p)
+                    if not ver:
+                        version = '<=0.1.2'
+                    else:
+                        version = '.'.join(
```

Comment:
Who provides this version object.  Would be very nice if this code was in there instead of here where it is messy.